### DISCUSSION TITLE
Multi-Project Database Sync

### DISCUSSION THREAD

**Discussion by fairhat** [blitz-js/blitz#2645] [Q&A]
Hello,

is there a recommended way to sync prisma types across multiple projects?
I can live with the fact that queries and mutations will still be separated. However, being able to sync my prisma schema would be nice.

```blitz p db pull && blitz p generate``` does work to get the schema for dev

However, tests fail and there seems to be no command to pull the test db?

If I change the schema on one project, the migrations folder will be out of sync. There seems to be no way to synchronize them?

**Comment by flybayer** [comment]
If your apps are in a monorepo, you can have separate package for prisma, schema, and migrations.

If they are in different repos, you could look into using git submodules, or [git subrepo](https://github.com/ingydotnet/git-subrepo)

**Reply by MrLeebo** [reply]
One thing to consider about these solutions is that if you are setting up a project specifically to contain your database client layer, it doesn't need to be blitz; you can set it up with prisma and call `npx prisma generate` directly. Then export the client code it generates as a package that the other projects can import. 

The drawback to this strategy of separating your DB layer from the app is that you will have a very difficult time making breaking changes to your database client. You probably would not want to export the raw prisma types, instead write wrapper functions that deal with primitives or your own defined types, so that you can change how the db schema is implemented without causing major downstream changes to the other projects.
